Improve SSR doc for Next.js and Remix (#6154)
adobe/react-spectrum: packages/dev/docs/pages/react-spectrum/ssr.mdx

[Next.js](https://nextjs.org) is a framework for building websites and web applications with React. It supports both server side rendering as well as static rendering. A small amount of configuration is required to get React Spectrum’s CSS working with Next.js.

Add the following to your `next.config.js` file. This will ensure that React Spectrum’s CSS is loaded properly by Next.js. Note that packages may need to be added to the config below depending on your version of React Spectrum.

```tsx
module.exports = {
  transpilePackages: [
    "@adobe/react-spectrum",

>     // add required @react-spectrum/<name> packages, such as:

Reviewer: I like how you added this as an option via a comment. There is another related PR ([replace package enumerations with glob-stars](https://github.com/adobe/react-spectrum/pull/6176/files#top)) that is using glob to manage these packages. Could we update your PR to use glob and `@react-spectrum/*`? It would automatically include all the packages without people needing to update their config.

Author: Done. (Thanks for mentioning that)

Reviewer: Thank you!

## Remix

[Remix](https://remix.run) is a full-stack React framework with nested routing.

To configure Remix to load React Spectrum styles, CSS Side-Effect Imports is required, and [Remix Vite](https://remix.run/docs/en/main/future/vite) has built-in support for it.

> However, to make it work with SSR, a small amount of configuration is required. Add the following to your `vite.config.js` or `vite.config.ts` file. This will ensure that React Spectrum’s CSS can be built properly. Note that packages may need to be removed or added to the config below if using an older or newer version of React Spectrum.

Reviewer 2: ```suggestion
However, to make it work with SSR, a small amount of configuration is required. Add the following to your `vite.config.js` or `vite.config.ts` file. This will ensure that React Spectrum’s CSS can be built properly. Note that packages may need to be added to the config below depending on your version of React Spectrum.
```

## Remix

[Remix](https://remix.run) is a full-stack React framework with nested routing.

To configure Remix to load React Spectrum styles, CSS Side-Effect Imports is required, and [Remix Vite](https://remix.run/docs/en/main/future/vite) has built-in support for it.

However, to make it work with SSR, a small amount of configuration is required. Add the following to your `vite.config.js` or `vite.config.ts` file. This will ensure that React Spectrum’s CSS can be built properly. Note that packages may need to be removed or added to the config below if using an older or newer version of React Spectrum.

```tsx
export default defineConfig({
  ssr: {
    noExternal: [
      '@adobe/react-spectrum',
      '@react-spectrum/actionbar',

>       '@react-spectrum/actiongroup',

Reviewer 2: I think it's ok to give the first couple, but after that, we should just make it a
```
... // fill in the rest
```

otherwise we'd need a way to auto update this list during release or something. we've had lists like this before and they get stale pretty quickly.

Author: Done.
